File: movie-ui/app/api/v1/auth/admin-login/route.ts
import { type NextRequest, NextResponse } from "next/server"

export async function POST(request: NextRequest) {
  try {
    const body = await request.json()
    const { email, password } = body

    // Validate input
    if (!email || !password) {
      return NextResponse.json({ error: "Email and password are required" }, { status: 400 })
    }

    // Simple hard-coded admin authentication
    if (email === "admin@example.com" && password === "admin123") {
      return NextResponse.json({
        success: true,
        user: {
          id: 999,
          email: "admin@example.com",
          role: "admin",
        },
        token: "admin-jwt-token-" + Math.random().toString(36).substring(2),
      })
    }

    // If credentials don't match
    return NextResponse.json({ error: "Invalid admin credentials" }, { status: 401 })
  } catch (error: any) {
    console.error("Admin login error:", error)
    return NextResponse.json({ error: "Internal server error" }, { status: 500 })
  }
}

File: flask/api/v1/auth/admin_login/route.py
from flask import Blueprint, request, jsonify
from models import Users  # Changed from User to Users (DDL-first model)
from serializers import ModelSerializer
from extensions import db
from werkzeug.security import check_password_hash
from sqlalchemy import select

admin_login_bp = Blueprint('admin_login', __name__)

@admin_login_bp.route('/', methods=['POST'])
def admin_login():
    """Admin login endpoint"""
    data = request.json
    
    if not data or not data.get('email') or not data.get('password'):
        return jsonify({'error': 'Email and password are required'}), 400
    
    user = db.session.execute(
        select(Users).where(Users.email == data['email'])
    ).scalar_one_or_none()
    
    if not user or user.role != 'admin':
        return jsonify({'error': 'Invalid admin credentials'}), 401
    
    if check_password_hash(user.password_hash, data['password']):
        return jsonify({
            'status': 'success',
            'data': ModelSerializer.serialize_users(user)
        }), 200
    
    return jsonify({'error': 'Invalid admin credentials'}), 401